src/transformers/generation_utils.py

```diff
@@ -300,6 +305,7 @@ def generate(
         num_return_sequences: Optional[int] = None,
         decoder_start_token_id: Optional[int] = None,
         use_cache: Optional[bool] = None,
+        prefix_allowed_tokens_fn: Callable = None,
```

Comment:
We should improve the type hinting here, either:

`Callable[[int, torch.Tensor], List[List[int]]` or `Callable[[torch.Tensor], List[List[int]]` if I understand correctly.